src/mixins/MemoryMixin.h

```diff
@@ -30,7 +30,7 @@ namespace reactivesocket {
 /// implicitly specified as virtual, depending on whether the Base class
 /// is implementing the (virtual) methods of the
 /// Subscription or the Subscriber interface.
-template <typename Base>
+template <typename Base, typename Payload = Payload>
```

Comment:
Sure `typename T = Payload`.  The issue is on line 71 below where we have

  void onNext(Payload payload) {

I want `Payload` to be generic.   `typename T = Payload` might be more explicit and then I'll change `Payload` elsewhere.